So, you just went to pick up your RNA-Seq results and the people at the sequencing facility handed you a hard drive full of ridulously named folders and .bcl files. What now?

The guides within this repository aim to walk you through identifying the files that you need, how to align those files to the genome, and then how to learn about the expression of RNA in your samples (including their differential expression between sample groups).

Please note that these guides are aimed at users at the University of Miami using the LSF-based computing cluster Pegasus, but most of the information presented here extends to any RNA-Seq processing situation. The files of code provided may not work for other computing clusters though. 

The subsections -- and what they try to explain -- are as follows:

# Finding the fastq files
  this section explains the data generated by a typical illumina next-gen sequencing machine (such as a HiSeq 2500) and its processing pipeline and which files you will be using for further analysis.

# Aligning the reads to the genome
  this section walks you through the installation and usage of the aligner Tophat

# Quantifying RNA expression
  this section discusses two main types of approaches to quantifying the RNA expression contained in RNA-Seq data -- gene-level and transcript-level quantification. It then walks through a typical pipeline for each. 
